src/server/scripts/Northrend/Naxxramas/boss_faerlina.cpp

```diff
@@ -147,102 +135,34 @@ class boss_faerlina : public CreatureScript
-                        if (Creature* cr = pInstance->instance->GetCreature(*itr))
-                        {
-                            if (cr->IsInCombat())
-                                DoZoneInCombat();
-                        }
-                    }
-                }
-            }
-
-            if (!UpdateVictim())
-                return;
-
-            events.Update(diff);
-            if (me->HasUnitState(UNIT_STATE_CASTING))
-                return;
-
-            switch (events.ExecuteEvent())
-            {
-                case EVENT_POISON_BOLT:
-                    if (!me->HasAura(RAID_MODE(SPELL_WIDOWS_EMBRACE, SPELL_MINION_WIDOWS_EMBRACE)))
-                    {
-                        me->CastCustomSpell(RAID_MODE(SPELL_POISON_BOLT_VOLLEY_10, SPELL_POISON_BOLT_VOLLEY_25), SPELLVALUE_MAX_TARGETS, RAID_MODE(3, 10), me, false);
-                    }
-                    events.Repeat(7s, 15s);
-                    break;
-                case EVENT_RAIN_OF_FIRE:
-                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
-                    {
-                        me->CastSpell(target, RAID_MODE(SPELL_RAIN_OF_FIRE_10, SPELL_RAIN_OF_FIRE_25), false);
-                    }
-                    events.Repeat(8s, 18s);
-                    break;
-                case EVENT_FRENZY:
-                    if (!me->HasAura(RAID_MODE(SPELL_FRENZY_10, SPELL_FRENZY_25)))
-                    {
-                        Talk(SAY_FRENZY);
-                        Talk(EMOTE_FRENZY);
-                        me->CastSpell(me, RAID_MODE(SPELL_FRENZY_10, SPELL_FRENZY_25), true);
-                        events.Repeat(1min);
-                    }
-                    else
-                    {
-                        events.Repeat(30s);
-                    }
-                    break;
-            }
-            DoMeleeAttackIfReady();
+            if (GameObject* go = me->GetMap()->GetGameObject(instance->GetGuidData(DATA_FAERLINA_WEB)))
+                go->SetGoState(GO_STATE_ACTIVE);
         }
 
         void SpellHit(Unit* caster, SpellInfo const* spell) override
         {
             if (spell->Id == RAID_MODE(SPELL_WIDOWS_EMBRACE, SPELL_MINION_WIDOWS_EMBRACE))
             {
                 Talk(EMOTE_WIDOWS_EMBRACE);
-                if (me->HasAura(RAID_MODE(SPELL_FRENZY_10, SPELL_FRENZY_25)))
-                {
-                    me->RemoveAurasDueToSpell(RAID_MODE(SPELL_FRENZY_10, SPELL_FRENZY_25));
-                    events.RescheduleEvent(EVENT_FRENZY, 1min);
-                }
-                pInstance->SetData(DATA_FRENZY_REMOVED, 0);
+                scheduler.RescheduleGroup(GROUP_FRENZY, 1min);
+                me->RemoveAurasDueToSpell(SPELL_FRENZY);
+                instance->SetData(DATA_FRENZY_REMOVED, 0);
                 if (Is25ManRaid())
-                {
-                    Unit::Kill(caster, caster);
-                }
+                    caster->KillSelf();;
```

Comment:
```suggestion
                    caster->KillSelf();
```

Reply:
we should have a CI for this

Reply:
we do have a CI for this

Reply:
https://github.com/azerothcore/azerothcore-wotlk/pull/21388 will fix catching this